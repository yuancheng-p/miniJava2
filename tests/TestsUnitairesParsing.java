package tutu.titi.toto;

class I {}

public class A {
    int a1 = 1;
    // int[] a = { 3,4,5 }; // not supported yet
    boolean a2 = false;
    String a6 = "jsdjkndjk\njdsdjkf";
}
abstract class B extends C {
    B() { super(); }
    private A mthis () throws tuty.A, tre.tr.b {
	int a, b;
	{
	    int c = 18;
	    { ; }
	    new A();
	    A.m();
	    a[2].m();
	    a[2][6].m();
	    B.A.m();
	    this.m();
	    super();
	    this();
	    super.m();
	    m();
	    for (int c; c > 6 ; c++) { }
	}
	//tutu.tyty.a = 1;
	//final tutu.tyty.a op;
	a(B.this);
	a = b = 1;
	return new A();
	try {
	} catch (E e) {
	    toto;
	} finally {
	    tutu;
	}
	try {
	} finally {
	    tutu;
	}
	try {
	} catch (E e) {
	    toto;
	}
    }
    int ab=1;
    A class_a;
    Int matt4() {
        this.ab;
        class_a.a1=2;
    }
}
static private class C extends A {}

