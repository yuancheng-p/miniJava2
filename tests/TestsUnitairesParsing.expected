successfull parsing
package tutu.titi.toto
class I extends Object {
}

public class A extends Object {
  int a1 = 1;
  boolean a2 = false;
  String a6 = "jsdjkndjk\njdsdjkf";
}

abstract class B extends C {
  int ab = 1;
  A class_a;
  B() {
  }
  A mthis() tuty.A,tre.tr.b {
    int a;
    int b;
    {
      int c 18;
      {
        ;
      }
      new A();
      this.m();
      m();
      for (int c; c>6;c++) {
        {
        }
      }
    }
    a=b=1;
    return new A();
    try {
    } catch (E e)
      toto;
    } finally {
      tutu;
    }
    try {
    } finally {
      tutu;
    }
    try {
    } catch (E e)
      toto;
    }
  }
  Int matt4()  {
    this.ab;
    class_a.a1=2;
  }
  void m()  {
  }
}

static private class C extends A {
}

start typing
check add and methods:B:mthis
check add and methods:B:matt4
check add and methods:B:m
--------- env --------
A : class_env(parent=: Object)
 attr:  String a6 = "jsdjkndjk\njdsdjkf";
 attr:  boolean a2 = false;
 attr:  int a1 = 1;
B : class_env(parent=: C)
method : matt4
method : m
method : mthis
 attr:  A class_a;
 attr:  int ab = 1;
Boolean : class_env(parent=: Object)
C : class_env(parent=: A)
I : class_env(parent=: Object)
Integer : class_env(parent=: Object)
Object : class_env(parent=: )
String : class_env(parent=: Object)
450450end typing
---------p c 
456
0---------p c 
has constructor
---------p c 
---------p c
